Source/WebCore/platform/network/HTTPParsers.cpp

```diff
@@ -610,20 +610,31 @@ OptionSet<ClearSiteDataValue> parseClearSiteDataHeader(const ResourceResponse& r
     return result;
 }
 
-bool parseRange(StringView range, long long& rangeOffset, long long& rangeEnd, long long& rangeSuffixLength)
+// Implements <https://fetch.spec.whatwg.org/#simple-range-header-value>.
+bool parseRange(StringView range, RangeAllowWhitespace allowWhitespace, long long& rangeStart, long long& rangeEnd)
 {
-    // The format of "Range" header is defined in RFC 2616 Section 14.35.1.
-    // http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.35.1
-    // We don't support multiple range requests.
+    rangeStart = rangeEnd = -1;
 
-    rangeOffset = rangeEnd = rangeSuffixLength = -1;
+    // FIXME: this whole function could be more efficient by walking through the range value once.
+    if (allowWhitespace == RangeAllowWhitespace::No) {
+        // Only 0x20 and 0x09 matter as newlines are already gone by the time we parse a header value.
+        if (range.find(' ') != notFound)
+            return false;
+        if (range.find('\t') != notFound)
+            return false;
```

Comment:
Would the following work:
```
static bool isTabOrSpace(UChar value)
{
    return value == '\t'|| value == ' ';
}

bool parseRange(...)
{
    ....
    if (allowWhitespace == RangeAllowWhitespace::No && range.find(isTabOrSpace) != notFound)
        return false;
}
```

Reply:
I will try that, but what I meant with my FIXME was turning `parseRange` more into something like https://fetch.spec.whatwg.org/#simple-range-header-value where you only iterate through the whole range once. That would be a lot easier if the `skipWhile` methods took a `StringView`.